crates/sui-adapter/src/adapter.rs

```diff
@@ -162,12 +156,20 @@ pub fn execute<
         by_value_objects,
         mutable_ref_objects,
         has_ctx_arg,
-    } = resolve_and_type_check(&objects, &module, function, &type_args, args, is_genesis)?;
+    } = resolve_and_type_check(
+        &objects,
+        &module,
+        function,
+        &type_args,
+        args,
+        is_genesis,
+        Mode::allow_arbitrary_function_calls(),
```

Comment:
Nit: could we pass in the entire `Mode` here rather than reading a bool? I am paranoid about accidentally passing the wrong boolean literal somewhere, whereas passing the wrong `Mode` seems like a hard mistake to make.